classes/file_search.php

```diff
@@ -151,17 +151,20 @@ public static function get_criteria(files $record): object {
      *
      * @param \tool_advancedreplace\files $record
      * @param string $output path
-     * @param int $limitfrom limit for sql
-     * @param int $limitnum limit for sql
+     * @param int $startid minimum id for sql
+     * @param int $endid maximum id for sql
+     * @param bool $finalshard True if this is the ladt shard
```

Comment:
Is finalshard needed? Logically I think it makes more sense to set endid to 0 for the final shard - since having it doesn't make sense if you're not going to use it. 

With that change it having a startid without an endid would be the indicator that it's the final shard.

Reply:
Well I think this is a program style question.  My preference is to code explicitly, as I have done.  Over-loading the endid to have a special meaning when it is null or zero is a bit obscure. 
With the finalshard parameter and variable it is obvious to all what we are intending.

Reply:
That's fair. 

I think the main point I was trying to make is more that we shouldn't be setting an endid if we're not going to use it - which I'd argue is actually less explicit. I can agree with the confusion about the default being 0 now that this has been changed to endid though. 

Anyway it's not worth going into this too deeply, I'm OK with keeping it if you have a strong preference.